src/hidpp20.h

```diff
@@ -389,26 +382,46 @@ enum hidpp20_led_type {
 	HIDPP20_LED_SIDE,
 };
 
+enum hidpp20_led_mode {
+	HIDPP20_LED_OFF = 0x00,
+	HIDPP20_LED_ON = 0x01,
+	HIDPP20_LED_CYCLE = 0x03,
+	HIDPP20_LED_BREATHING = 0x0a,
+};
+
+enum hidpp20_led_waveform {
+	HIDPP20_LED_WF_DEFAULT = 0x00,
+	HIDPP20_LED_WF_SINE = 0x01,
+	HIDPP20_LED_WF_SQUARE = 0x02,
+	HIDPP20_LED_WF_TRIANGLE = 0x03,
+	HIDPP20_LED_WF_SAWTOOTH = 0x04,
+	HIDPP20_LED_WF_SHARKFIN = 0x05,
+	HIDPP20_LED_WF_EXPONENTIAL = 0x06,
+};
+
 struct hidpp20_internal_led {
-	uint8_t mode; /* 00 - off; 01 - on; 03 - cycle; 0a - breath */
-	struct hidpp20_color color;
+	uint8_t mode; /* enum hidpp20_led_mode */
 	union {
+		struct hidpp20_led_fixed {
+			struct hidpp20_color color;
+			uint8_t effect;
+		} __attribute__((packed)) fixed;
 		struct hidpp20_led_cycle {
-			uint16_t pad_1;
-			uint16_t rate;  /* 16000 - 20000Hz */
-			uint8_t brightness; /* 0 - 100 percent */
-			uint16_t pad_2;
+			uint8_t unused[5];
+			uint16_t rate_or_speed;  /* 16000 - 20000Hz */
+			uint8_t intensity; /* 1 - 100 percent, 0 means 100 */
 		} __attribute__((packed)) cycle;
 		struct hidpp20_led_breath {
-			uint16_t rate;
-			uint8_t pad_1;
-			uint8_t brightness;
-			uint8_t pad_2;
-			uint16_t pad_3;
+			struct hidpp20_color color;
+			uint16_t speed_or_period;
```

Comment:
Thanks for the clarification. I guess getZoneEffectInfo is function 2 and is not currently implemented. It looks coherent with what you said.